lua/astrocommunity/completion/cmp-cmdline/init.lua

```diff
@@ -1,22 +1,31 @@
-local cmp = require "cmp"
+local cmp_mapping = require "cmp.config.mapping"
+local cmp_sources = require "cmp.config.sources"
 
 return {
-  "hrsh7th/cmp-cmdline",
-  lazy = false,
-  opts = {
-    mapping = cmp.mapping.preset.cmdline(),
-    sources = cmp.config.sources({
-      { name = "path" },
-    }, {
-      {
-        name = "cmdline",
-        option = {
-          ignore_cmds = { "Man", "!" },
+  {
+    "hrsh7th/cmp-cmdline",
+    opts = {
+      mapping = cmp_mapping.preset.cmdline(),
+      sources = cmp_sources({
+        { name = "path" },
+      }, {
+        {
+          name = "cmdline",
+          option = {
+            ignore_cmds = { "Man", "!" },
+          },
         },
-      },
-    }, {
-      { name = "buffer" },
-    }),
+      }, {
+        { name = "buffer" },
+      }),
+    },
+    config = function(_, opts) require("cmp").setup.cmdline(":", opts) end,
+  },
+  {
+    "nvim-cmp",
+    event = { "CmdlineEnter" },
+    dependencies = {
+      "cmp-cmdline",
+    },
   },
```

Comment:
Why do you have this?

Reply:
Otherwise, the `cmp-cmdline` won't get loaded.

Reply:
I believe that's the incorrect way to do it.
Instead, you should make `nvim-cmp` a dependency of `cmp-cmdline`, and make `cmp-cmdline` load on CmdLineEnter.

Reply:
fixed.